flow/connectors/postgres/postgres.go

```diff
@@ -57,6 +60,33 @@ type PostgresConnector struct {
 	pgVersion              shared.PGVersion
 }
 
+func ParseConfig(connectionString string, pgConfig *protos.PostgresConfig) (*pgx.ConnConfig, error) {
+	connConfig, err := pgx.ParseConfig(connectionString)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse connection string: %w", err)
+	}
+	if pgConfig.RootCa != nil {
+		caPool := x509.NewCertPool()
+		if !caPool.AppendCertsFromPEM([]byte(*pgConfig.RootCa)) {
+			return nil, errors.New("failed to parse provided root CA")
+		}
+
+		connConfig.TLSConfig = &tls.Config{
+			MinVersion: tls.VersionTLS12,
+			RootCAs:    caPool,
+		}
+		if net.ParseIP(connConfig.Host) == nil {
```

Comment:
can we move these to a file called shared/certs.go and combine the logic for MySQL and Postgres.